app/timetables/models.py

```diff
@@ -106,11 +106,11 @@ class Timetable(SlugifyMixin, TimestampMixin):
     slugify_field = 'name'
 
     def clean(self):
-        # Ensure current_cycle_day and cycle_length are not None before compare
-        if self.current_cycle_day and self.cycle_length:
-            if self.current_cycle_day > self.cycle_length:
+        # Ensure ref_cycle_day and cycle_length are not None before compare
```

Comment:
Is it ever possible that these would have a value of `None`; considering that they are both instances of a class ?

Is the check here really necessary ?